guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md

```diff
@@ -0,0 +1,355 @@
+```
+// (1):
+buildscript {
+  repositories {
+    maven { url "https://dl.bintray.com/spinnaker/gradle/" }
+  }
+  dependencies {
+    classpath("com.netflix.spinnaker.gradle:spinnaker-extensions:$spinnakerGradleVersion")
+  }
+}
+
+// (2):
+plugins {
+  //... other plugins you might be using
+  id("com.moowork.node").version("1.3.1").apply(false)
+}
+
+// (3):
+apply plugin: "io.spinnaker.plugin.bundler"
+spinnakerBundle {
+  pluginId = "Armory.RandomWaitPlugin"
+  description = "An example of a PF4J based plugin, that provides a new stage."
+  provider = "https://github.com/spinnaker-plugin-examples"
+  version = rootProject.version
+}
+```
+
+**Notes:**
+1. The `buildscript` block brings in the spinnaker-extensions gradle tooling, which includes
+   plugins for creating a bundle, ui-extension, and service-extension.
+2. The `plugins` block imports the node plugin but does not apply it. This plugin is needed by
+   the ui-extension to build the us assets.
+3. At the top level we apply the `io.spinnaker.plugin.bundler` to define the name of the plugin
+   bundle this project is producing along with bundle metadata.
+
+### ui-extension build.gradle
+
+To build a UI extension with the base conventions, the only thing necessary is
+applying the `io.spinnaker.plugin.ui-extension` plugin.
+
+**TODO** document what the ui-extension plugin supplies / supports. Maybe some of that is below?
+
+```
+apply plugin: "io.spinnaker.plugin.ui-extension"
+```
+
+### service-extension build.gradle
+Here are the basic things your service-extension gradle build needs. Note that the example project
+includes a bunch of additional configuration to configure kotlin for the plugin. The configuration
+below strips that out to just include the basic gradle configuration necessary for a JVM plugin, if
+you wish to develop plugins in kotlin refer to the gradle build from the example project.
+
+```
+// (1):
+apply plugin: "io.spinnaker.plugin.service-extension"
+apply plugin: "maven-publish"
+
+// (2):
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+
+dependencies {
+  // (3):
+  annotationProcessor("org.pf4j:pf4j:$pf4jVersion)
+
+  // (4):
+  compileOnly("com.netflix.spinnaker.kork:kork-plugins-api:$korkVersion")
+  compileOnly("com.netflix.spinnaker.orca:orca-api:$orcaVersion")
+
+  // (5):
+  // implementation("com.something.my.plugin.needs:dependency:version")
+}
+
+// (6):
+spinnakerPlugin {
+  serviceName = "orca"
+  pluginClass = "io.armory.plugin.stage.wait.random.RandomWaitPlugin"
+}
+```
+
+**Notes:**
+1. The `io.spinnaker.plugin.service-extension` and `maven-publish` are used for plugin bundling
+   and metadata generation.
+2. Spinnaker services require JDK 1.8 compatible bytecode.
+3. Enable pf4j annotation processing to build plugin extension metadata
+4. The plugin APIs are marked as `compileOnly` so as to not include them in the resulting plugin
+   jar. These dependencies must be loaded from the service's classloader, not the plugin classloader.
```

Comment:
> These dependencies must be loaded from the service's classloader, not the plugin classloader.

This seems like something that would be a better fit for internals docs, rather than a getting started doc. The way it's worded here I think would lead to confusion: To me, it reads like It's hinting at an action that developers must take, but there really isn't any action to take on their part.

Reply:
changed "must be" to "are"